### DISCUSSION TITLE
using alevin-fry in Nextflow pipeline nf-core/scrnaseq?

### DISCUSSION THREAD

**Discussion by Khajidu** [COMBINE-lab/alevin-fry#52] [Q&A]
I heard alevin-fry would soon replace alevin in the Nextflow pipeline nf-core/scrnaseq, but how can I use it in the meantime? I have data that look like 10XV1 data (1 index file and 2 read files for each read pair) for info (I cannot share them unfortunately).

**Comment by rob-p** [comment]
Hi @Khajidu,

  Thanks for bringing this up. What organism are you working in? Is your data a standard 10x v1 library? If so, we can demonstrate the mock commands with another public dataset that should then work with your data. @DongzeHE, could we please put together a small set of commands that will run some 10x v1 data through the alevin-fry pipeline?

Thanks!
Rob

**Comment by Khajidu** [comment]
I'm working on catshark brain samples, the data has been sequenced with 10XV3 technology, but is organized in 3 files per read pair as described above, as if it were 10XV1 data. Treating it as V3 data gives no mappings.

**Reply by rob-p** [reply]
That is quite strange!  Do you know the organization of where, in the index file, the CB or UMI resides?  This seems like a _custom_ library setup, as the standard Chromium v3 does not make use of the index reads for CB or UMI information (unlike the v1 data).  However, if we know the location of the CB / UMI on the reads, we can process the data, in a streaming way, into a format that can be properly parsed by the pipeline.

**Reply by Khajidu** [reply]
More precisely: Chromium Next GEM Single Cell 3ʹ Reagent Kits v3.1 techonology, details here https://assets.ctfassets.net/an68im79xiti/1eX2FPdpeCgnCJtw4fj9Hx/7cb84edaa9eca04b607f9193162994de/CG000204_ChromiumNextGEMSingleCell3_v3.1_Rev_D.pdf

**Reply by rob-p** [reply]
Thanks!  But this document describes the CB and UMI as belonging to R1:

<img width="537" alt="image" src="https://user-images.githubusercontent.com/361470/166942027-3732409d-33c2-4f35-9151-3611be3fc2f8.png">

Do you know specifically how your library is laid out (i.e. where on the specific reads the CB and UMI reside, and their length)?

**Reply by Khajidu** [reply]
I have 8-bp sequences in I1, and short (28-bp) reads in R1, looks like the cutoff is between barcode (8-bp?) and UMI. I asked our collaborators for confirmation.

**Reply by DongzeHE** [reply]
Hello @Khajidu, would you mind helping me to understand your situation better? 

So the I1 file is from Illumina and is only used for demultiplexing sequencing runs for Illumina. It is not used in the alevin-fry pipeline. The 28-bp reads in R1 seem actually in their expected size, i.e., 16-nt 10x Barcode + 12-nt UMI. 

Therefore, as you said that you have no mappings if treating your data as a V3 experiment, I would like first to figure out whether some rare bug in our code causes this, your data is really in a strange format, or this is because the salmon index or the command you used for running salmon. 

If possible, could you please share the code you used for your V3 attempt?


Best,
Dongze

**Reply by Khajidu** [reply]
I get no mappings even with other aligners, this is why I wanted to try alevin-fry (I only used alevin, star and kallisto)

**Reply by DongzeHE** [reply]
Unfortunately, alevin and alevin-fry both are based on [pufferfish](https://github.com/COMBINE-lab/pufferfish), so if you get no mappings with alevin, I would not think running alevin-fry will make any difference. Instead, do you think that there is any chance that the reference transcriptome used for building salmon index has something wrong? 

If possible, could you please share the commands you used for building your salmon index?

**Reply by rob-p** [reply]
So, just for some context @DongzeHE — these other tools were run through the `nf-core` scrnaseq pipeline.  So you can find the exact commands used there.

It's possible mapping issues are due to the reference or data quality etc., but 0 is a *very low* mapping rate.  If the library preparation is, in fact, non-standard though, that would perhaps explain what is going on.

**Reply by Khajidu** [reply]
In fact, I get a non-zero mapping rate... over 0 sequences (I work from a genome and a GFF and it gives me 50k sequences, but it seems that none of them is being mapped on)

**Reply by rob-p** [reply]
Right, sorry — what I mean by mapping rate here is the number of sequenced reads from your experiment that are assigned to some target transcript.  However, in all of the single-cell pipelines, it's possible to have a non-zero mapping rate, but still not have valid cells identified due to the way valid cell barcodes are determined, filtered, etc.

**Reply by Khajidu** [reply]
oh that would be why... I'll be looking into this tomorrow.
